crypto/chacha/asm/chacha-x86_64.pl

```diff
@@ -107,24 +112,16 @@
 .asciz	"ChaCha20 for x86_64, CRYPTOGAMS by <appro\@openssl.org>"
 ___
 
-sub AUTOLOAD()          # thunk [simplified] 32-bit style perlasm
-{ my $opcode = $AUTOLOAD; $opcode =~ s/.*:://;
-  my $arg = pop;
-    $arg = "\$$arg" if ($arg*1 eq $arg);
-    $code .= "\t$opcode\t".join(',',$arg,reverse @_)."\n";
-}
-
-@x=("%eax","%ebx","%ecx","%edx",map("%r${_}d",(8..11)),
-    "%nox","%nox","%nox","%nox",map("%r${_}d",(12..15)));
-@t=("%esi","%edi");
+my @x=("%eax","%ebx","%ecx","%edx",map("%r${_}d",(8..11)),
+       "%nox","%nox","%nox","%nox",map("%r${_}d",(12..15)));
+my @t=("%esi","%edi");
 
 sub ROUND {			# critical path is 24 cycles per round
 my ($a0,$b0,$c0,$d0)=@_;
 my ($a1,$b1,$c1,$d1)=map(($_&~3)+(($_+1)&3),($a0,$b0,$c0,$d0));
 my ($a2,$b2,$c2,$d2)=map(($_&~3)+(($_+1)&3),($a1,$b1,$c1,$d1));
 my ($a3,$b3,$c3,$d3)=map(($_&~3)+(($_+1)&3),($a2,$b2,$c2,$d2));
-my ($xc,$xc_)=map("\"$_\"",@t);
-my @x=map("\"$_\"",@x);
+my ($xc,$xc_)=@t;
```

Comment:
I agree perl's `$x[0]` syntax is rather silly and wish it were `@x[0]` and `%x{blah}`. But perl's syntax is what it is. I think, where there isn't a good perlasm-specific reason to deviate from perl conventions, perlasm should follow them. This code needs to get to the point that it is understandable by other people, and removing surprises like this is part of this.